plugin/src/main/java/org/wildfly/plugin/common/Utils.java

```diff
@@ -101,4 +118,66 @@ public static List<String> splitArguments(final CharSequence arguments) {
         }
         return args;
     }
+
+    public static ScanResults scanDeployment(GlowConfig discoverProvisioningInfo,
+            List<String> layers,
+            List<String> excludedLayers,
+            List<FeaturePack> featurePacks,
+            boolean dryRun,
+            Log log,
+            Path deploymentContent,
+            MavenRepoManager artifactResolver,
+            Path outputFolder,
+            ProvisioningManager pm,
+            Map<String, String> galleonOptions,
+            String layersConfigurationFileName) throws Exception {
+        if (!layers.isEmpty()) {
+            throw new MojoExecutionException("layers must be empty when enabling glow");
+        }
+        if (!excludedLayers.isEmpty()) {
+            throw new MojoExecutionException("excluded layers must be empty when enabling glow");
+        }
+        if (!Files.exists(deploymentContent)) {
+            throw new MojoExecutionException("A deployment is expected when enabling glow layer discovery");
+        }
+        Path inProvisioningFile = null;
+        Path glowOutputFolder = outputFolder.resolve("glow-scan");
+        Files.createDirectories(glowOutputFolder);
+        if (!featurePacks.isEmpty()) {
+            ProvisioningConfig in = GalleonUtils.buildConfig(pm, featurePacks, layers, excludedLayers, galleonOptions,
+                    layersConfigurationFileName);
+            inProvisioningFile = glowOutputFolder.resolve("glow-in-provisioning.xml");
+            try (FileWriter fileWriter = new FileWriter(inProvisioningFile.toFile())) {
+                ProvisioningXmlWriter.getInstance().write(in, fileWriter);
+            }
+        }
+        Arguments arguments = discoverProvisioningInfo.toArguments(deploymentContent, inProvisioningFile);
+        log.info("Glow is scanning... ");
+        ScanResults results;
+        GlowMavenMessageWriter writer = new GlowMavenMessageWriter(log);
+        try {
+            results = GlowSession.scan(artifactResolver, arguments, writer);
+        } catch (Exception ex) {
+            throw new MojoExecutionException(ex.getLocalizedMessage(), ex);
+        }
+
+        log.info("Glow scanning DONE.");
+        try {
+            results.outputInformation(writer);
+        } catch (Exception ex) {
+            throw new MojoExecutionException(ex.getLocalizedMessage(), ex);
+        }
+        if (!dryRun) {
+            results.outputConfig(glowOutputFolder, false);
+        }
+        if (results.getErrorSession().hasErrors()) {
+            if (discoverProvisioningInfo.isFailsOnError()) {
+                throw new MojoExecutionException("Error detected by glow. Aborting.");
```

Comment:
Will errors be printed somewhere? What I mean by this, is will glow log some kind of error?

Reply:
Yes, glow has a pluggable message writer tha , in the plugin cases route logs to maven logs: https://github.com/jfdenise/wildfly-maven-plugin/blob/glow-only/plugin/src/main/java/org/wildfly/plugin/common/GlowMavenMessageWriter.java
BTW I am changing glow by "WildFly Glow"